diags/diags.go

```diff
@@ -16,182 +16,51 @@ package diags
 
 import (
 	"fmt"
-	"sort"
 	"strings"
 )
 
 // A formatter for when a field or property is used that does not exist.
-type NonExistantFieldFormatter struct {
+type NonExistentFieldFormatter struct {
 	ParentLabel         string
 	Fields              []string
 	MaxElements         int
 	FieldsAreProperties bool
 }
 
-func (e NonExistantFieldFormatter) fieldsName() string {
+func (e NonExistentFieldFormatter) fieldsName() string {
 	if e.FieldsAreProperties {
 		return "properties"
 	}
 	return "fields"
 }
 
-// Get a single line message.
-func (e NonExistantFieldFormatter) Message(field, fieldLabel string) string {
-	return fmt.Sprintf("%s %s", e.messageHeader(fieldLabel), e.messageBody(field))
-}
-
-// A message broken up into a top level and detail line
-func (e NonExistantFieldFormatter) MessageWithDetail(field, fieldLabel string) (string, string) {
-	return e.messageHeader(fieldLabel), e.messageBody(field)
-}
-
-func (e NonExistantFieldFormatter) messageHeader(fieldLabel string) string {
+func (e NonExistentFieldFormatter) messageHeader(fieldLabel string) string {
 	return fmt.Sprintf("%s does not exist on %s", fieldLabel, e.ParentLabel)
 }
 
-func (e NonExistantFieldFormatter) messageBody(field string) string {
+func (e NonExistentFieldFormatter) messageBody(field string) string {
 	existing := sortByEditDistance(e.Fields, field)
 	if len(existing) == 0 {
 		return fmt.Sprintf("%s has no %s", e.ParentLabel, e.fieldsName())
 	}
 	list := strings.Join(existing, ", ")
 	if len(existing) > e.MaxElements && e.MaxElements != 0 {
-		list = fmt.Sprintf("%s and %d others", strings.Join(existing[:5], ", "), len(existing)-e.MaxElements)
-	}
-	return fmt.Sprintf("Existing %s are: %s", e.fieldsName(), list)
-}
-
-// A formatter for missing fields that may be valid in other places.
-type InvalidFieldBagFormatter struct {
-	ParentLabel string
-	Bags        []TypeBag
-	// The maximum number of fields to return. `MaxListed` <= 0 implies return all fields.
-	MaxListed int
-	// The degree of (edit) distance between the field typed and suggested.
-	DistanceLimit int
-}
-
-type TypeBag struct {
-	Name       string
-	Properties []string
-}
-
-func (e InvalidFieldBagFormatter) BagList(field string) []BagOrdering {
-	bags := suggestBag(field, e.Bags)
```

Comment:
Was this unused code?

Reply:
Yes, it was apparently copied from pulumi-yaml, but isn't used in esc